src/main/webapp/get-blur-areas.js

```diff
@@ -0,0 +1,235 @@
+                  alert('Unknown error while parsing results.' +
+                      '\nPlease reload the page and try again ' +
+                      'after some time.');
+                  reject(error);
+                }
+              },
+          );
+        },
+
+        (error) => {
+          alert('Unknown error while uploading image.' +
+              '\nPlease reload the page and try again after some time.');
+          reject(error);
+        },
+    );
+  });
+}
+
+/**
+ * Constructor for rectangle to blur.
+ * Takes as parameter rectangle from server - array with 4 points,
+ * point is an object with properties 'x' and 'y'.
+ * Validates rectangle from server.
+ * @param {Array<Object>} rect
+ * @param {Image} image
+ * @constructor
+ */
+function Rect(rect, image) {
+  if (!Array.isArray(rect)) {
+    throw new Error('Not a rectangle');
+  }
+
+  // has 4 points
+  if (rect.length !== 4) {
+    throw new Error('Not a rectangle');
+  }
+
+  // points must have x and y properties
+  for (const point of rect) {
+    if (!point.hasOwnProperty('x') || !point.hasOwnProperty('y')) {
+      throw new Error('Not a rectangle');
+    }
+  }
+
+  // points must not duplicate
+  for (let i = 0; i < rect.length; i++) {
+    for (let j = 0; j < rect.length; j++) {
+      if (i === j) {
+        continue;
+      }
+
+      if (rect[i].x === rect[j].x && rect[i].y === rect[j].y) {
+        throw new Error('Not a rectangle');
+      }
+    }
+  }
+
+  // get leftX, rightX, topY and bottomY
+  // as min or max values
+  this.leftX = rect[0].x;
+  this.topY = rect[0].y;
+  let rightX = rect[0].x;
+  let bottomY = rect[0].y;
+
+  for (const point of rect) {
+    this.leftX = Math.min(this.leftX, point.x);
+    this.topY = Math.min(this.topY, point.y);
+    rightX = Math.max(rightX, point.x);
+    bottomY = Math.max(bottomY, point.y);
+  }
+
+  // all point's x and y must equal minimum or maximum of those values.
+  // if points do not duplicate, it's 'x' has 2 options - either leftX
+  // or rightX, 'y' has 2 options, those options are not equal, and
+  // rect object has exactly 4 points,
+  // then we are sure, that this is a rectangle, parallel to x and y axes.
+  for (const point of rect) {
+    if (point.x !== this.leftX && point.x !== rightX) {
+      throw new Error('Not a rectangle');
+    }
+    if (point.y !== this.topY && point.y !== bottomY) {
+      throw new Error('Not a rectangle');
+    }
+  }
+
+  // minimum and maximum of x or y must not equal each other
+  if (this.leftX === rightX || this.topY === bottomY) {
+    throw new Error('Not a rectangle');
+  }
+
+  this.height = bottomY - this.topY + 1;
+  this.width = rightX - this.leftX + 1;
+
+  // rect must not have points outside the image
+  if (this.leftX < 0 || this.topY < 0) {
+    throw new Error('Rect has negative points');
+  }
+  if (bottomY > image.height || rightX > image.width) {
+    throw new Error('Rect is greater than image');
+  }
```

Comment:
There is no topY, bottomY, rightX and leftX variables in the beginning of a function. Although I can move this.height and this.width initialisation a bit more to the start, so that now the function will look like this:
checks;
initialisation of topY, bottomY, rightX, leftX, width and height;
checks that include the variables above;